test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/DomainPartGenerator.java

```diff
@@ -50,73 +48,60 @@
-     * @param Random to use for generation
+     *
+     * @param topLevelDomains top level domains to use for generation
      */
-    DomainPartGenerator(final Random random) {
-        this.random = random;
+    DomainPartGenerator(final List<String> topLevelDomains) {
+        this.topLevelDomains.addAll(topLevelDomains);
     }
 
     /**
      * Generate the domain part
-     * 
+     *
      * @return the domain part
      */
     public String next() {
-        return generateDomainPart();
-    }
-
-    /**
-     * Generate domain part
-     * 
-     * @return domain part
-     */
-    private String generateDomainPart() {
-        String generated = generateStringFromCharacters(random, DOMAINPART_CHARACTERS, MIN_LENGTH,
-                        MAX_LENGTH);
+        String generated = generateStringFromCharacters(DOMAIN_PART_CHARACTERS, MIN_LENGTH,
+                MAX_LENGTH);
         if (!passBasicChecks(generated)) {
-            return generateDomainPart();
+            return next();
         }
-        // toss 100 sided thing
-        if (isRandomlyTrue(100)) {
-            // toss coin
-            if (isRandomlyTrue(2)) {
-                generated = join("", "(comment)", generated);
+        generated = insertOptionalComment(generated);
+
+        generated = join(".", generated, values(topLevelDomains).next());
+
+        return generated;
+    }
+
+    private String insertOptionalComment(final String generated) {
+        String resultString = generated;
+        final boolean includeComment = BOOLEAN.next();
+        final boolean appendComment = BOOLEAN.next();
+
+        if (includeComment) {
+            if (appendComment) {
+                resultString = join("", generated, "(comment)");
             } else {
-                generated = join("", generated, "(comment)");
+                resultString = join("", "(comment)", generated);
             }
         }
-        generated = join(".", generated,
-                        topLevelDomains.get(chooseRandomPosition(topLevelDomains.size())));
-
-        return generated;
+        return resultString;
     }
 
     public List<String> getTopLevelDomains() {
         return topLevelDomains;
     }
 
-    public DomainPartGenerator setTopLevelDomains(final List<String> topLevelDomains) {
-        this.topLevelDomains.addAll(topLevelDomains);
-        return this;
-    }
-
     /**
      * Check that a domain part is valid
-     * 
-     * @param textToCheck
+     *
      * @return boolean value denoting a valid or invalid domain part
      */
     public static boolean passBasicChecks(final String textToCheck) {
         return !textToCheck.isEmpty() && !(MAX_LENGTH < textToCheck.length())
-                        && !textToCheck.startsWith("-") && !textToCheck.endsWith("-")
-                        && !textToCheck.matches("[0-9]+");
+                && !textToCheck.startsWith("-") && !textToCheck.endsWith("-")
+                && !textToCheck.matches("\\d+");
     }
```

Comment:
This is quite a bit of logic to test. i.e. writing tests that cover every possible path through this method.
Maybe better to pull this method out into it's own class and test all possibilities there.